test/unit/feirari/RewardsDistributorAdmin.test.ts

```diff
@@ -0,0 +1,114 @@
+import { expectRevert, balance, getAddresses, getCore } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers, artifacts } from 'hardhat';
+import { Signer } from 'ethers';
+import testHelpers, { BN, ether } from '@openzeppelin/test-helpers';
+
+const toBN = ethers.BigNumber.from;
+const {
+    constants: { ZERO_ADDRESS }
+} = testHelpers;
+
+const MockRewardsDistributor = artifacts.readArtifactSync('MockRewardsDistributor');
+const RewardsDistributorAdmin = artifacts.readArtifactSync('RewardsDistributorAdmin');
+
+describe('AutoRewardsDistributor', function () {
+  let governorAddress: string;
+  let pcvControllerAddress: string;
+
+  const e18 = '000000000000000000';
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [
+      addresses.governorAddress,
+      addresses.pcvControllerAddress,
+    ];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ pcvControllerAddress, governorAddress } = await getAddresses());
+
+    this.core = await getCore();
+    this.rewardsDistributor = await (await ethers.getContractFactory('MockRewardsDistributor')).deploy();
+    this.rewardsDistributorAdmin = await (await ethers.getContractFactory('RewardsDistributorAdmin'))
+        .deploy(this.core.address, this.rewardsDistributor.address);
+
+    await this.rewardsDistributor.transferOwnership(this.rewardsDistributorAdmin.address);
+  });
+
+  describe('Init', function () {
+    it('rewardsDistributorContract', async function () {
+      expect(await this.rewardsDistributorAdmin.rewardsDistributorContract()).to.be.equal(this.rewardsDistributor.address);
+    });
+
+    it('compSupplySpeeds', async function () {
+      expect(await this.rewardsDistributorAdmin.compSupplySpeeds(ZERO_ADDRESS)).to.be.equal(toBN('0'));
```

Comment:
what is this testing exactly?

Reply:
testing that initial compSupplySpeeds are 0. I pass 0 address as this doesn't matter in the test because it is just hitting a mock.